core/src/main/java/hudson/util/ProcessTree.java

```diff
@@ -604,7 +604,11 @@ public void killRecursively() throws InterruptedException {
                 PID_FIELD = clazz.getDeclaredField("pid");
                 PID_FIELD.setAccessible(true);
 
-                DESTROY_PROCESS = clazz.getDeclaredMethod("destroyProcess",int.class);
+                if (isPreJava8()) {
+                    DESTROY_PROCESS = clazz.getDeclaredMethod("destroyProcess",int.class);
+                } else {
+                    DESTROY_PROCESS = clazz.getDeclaredMethod("destroyProcess",int.class, boolean.class);
+                }
```

Comment:
Checking the Java version seems unnecessary and fragile. Could instead check for one overload of `destroyProcess`, and failing that, check for the other.

Throwing up a `LinkageError` is also inappropriate. A warning should be logged but then `UnixProcess.kill` should be a no-op. Otherwise the caller is totally broken, as in 1.554.3:

```
SEVERE: Failed to execute command close (channel …)
java.lang.LinkageError
	at hudson.util.ProcessTree$UnixReflection.<clinit>(ProcessTree.java:618)
	at hudson.util.ProcessTree$Unix.get(ProcessTree.java:485)
	at hudson.util.ProcessTree.killAll(ProcessTree.java:147)
	at hudson.slaves.CommandLauncher$1.onClosed(CommandLauncher.java:135)
	at hudson.remoting.Channel.terminate(Channel.java:799)
	at hudson.remoting.Channel$CloseCommand.execute(Channel.java:931)
	at hudson.remoting.Channel$2.handle(Channel.java:461)
	at hudson.remoting.SynchronousCommandTransport$ReaderThread.run(SynchronousCommandTransport.java:60)
Caused by: java.lang.NoSuchMethodException: java.lang.UNIXProcess.destroyProcess(int)
	at java.lang.Class.getDeclaredMethod(Class.java:2122)
	at hudson.util.ProcessTree$UnixReflection.<clinit>(ProcessTree.java:607)
	... 7 more
```